src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -49,6 +106,14 @@ private void handleCommandEntered() {
         }
     }
 
+    private void updateCommandFieldFromQueue() {
+        if (!commandQueue.isEmpty()) {
+            commandTextField.setText(commandQueue.peek());
+            commandTextField.end(); // Moves caret to end
```

Comment:
I did not find the workings of this line immediately obvious (I did not infer the implication of moving the caret to the end immediately), so I think it might be good for future reference to write some more comments here on what this line does. e.g. This clears the input **and** feeds it into the app by triggering `handleCommandEntered`.